src/lib/device/round_robin_chunk_selector.cpp

```diff
@@ -12,23 +12,22 @@
  * specific language governing permissions and limitations under the License.
  *
  *********************************************************************************/
-#include "device/chunk_selector.hpp"
-#include "device/chunk.h"
+#include "round_robin_chunk_selector.h"
 
 namespace homestore {
 RoundRobinChunkSelector::RoundRobinChunkSelector(bool dynamic_chunk_add) : m_dynamic_chunk_add{dynamic_chunk_add} {
     RELEASE_ASSERT_EQ(dynamic_chunk_add, false,
                       "Dynamically adding chunk to chunkselector is not supported, need RCU to make it thread safe");
 }
 
-void RoundRobinChunkSelector::add_chunk(cshared< Chunk >& chunk) { m_chunks.push_back(chunk); }
+void RoundRobinChunkSelector::add_chunk(VChunk chunk) { m_chunks.push_back(chunk); }
 
 Chunk* RoundRobinChunkSelector::select(blk_count_t, const blk_alloc_hints&) {
     if (*m_next_chunk_index >= m_chunks.size()) { *m_next_chunk_index = 0; }
-    return m_chunks[(*m_next_chunk_index)++].get();
+    return m_chunks[(*m_next_chunk_index)++].getInternalChunk().get();
 }
 
-void RoundRobinChunkSelector::foreach_chunks(std::function< void(cshared< Chunk >&) >&& cb) {
+void RoundRobinChunkSelector::foreach_chunks(std::function< void(VChunk&) >&& cb) {
```

Comment:
is there a reason why it can not be a shared_ptr?

Reply:
Vchunk has a shared_ptr in itself, so we do not need a shard_ptr of shared_ptr

Reply:
My original comment is again based on VChunk is an abstract class, which doesn't hold shared_ptr of Chunk. Nevermind if you want to keep it this way.